api/managementpb/agent.proto

```diff
@@ -0,0 +1,116 @@
+  string agent_password = 2;
+  // Agent type.
+  string agent_type = 3;
+  // AWS Access Key.
+  string aws_access_key = 4;
+  // AWS Secret Key.
+  string aws_secret_key = 5;
+  // Options used when adding an Azure exporter.
+  string azure_options = 6;
+  // Creation timestamp.
+  google.protobuf.Timestamp created_at = 7;
+  // Custom user-assigned labels.
+  map<string, string> custom_labels = 8;
+  // Desired Agent status: enabled (false) or disabled (true).
+  bool disabled = 9;
+  // List of disabled collector names.
+  repeated string disabled_collectors = 10;
+  // Listen port for scraping metrics.
+  uint32 listen_port = 11;
+  // Log level for exporter.
+  string log_level = 12;
+  // Limit query length in QAN.
+  int32 max_query_length = 13;
+  // Limit query log size in QAN.
+  int64 max_query_log_size = 14;
+  // Path under which metrics are exposed, used to generate URI.
+  string metrics_path = 15;
+  // Scheme to generate URI to exporter metrics endpoints.
+  string metrics_scheme = 16;
+  // TLS options for connecting to MongoDB.
+  string mongo_db_options = 17;
+  // Options for connecting to MySQL.
+  string mysql_options = 18;
+  // Node identifier.
+  string node_id = 19;
+  // Password for connecting the agent to the database.
+  string password = 20;
+  // The pmm-agent identifier which runs this instance.
+  string pmm_agent_id = 21;
+  // Options for connecting to PostgreSQL.
+  string postgresql_options = 22;
+  // Path to exec process.
+  string process_exec_path = 23;
+  // True if exporter uses push metrics mode.
+  bool push_metrics = 24;
+  // True if query examples are disabled.
+  bool query_examples_disabled = 25;
+  // True if RDS basic metrics are disdabled.
+  bool rds_basic_metrics_disabled = 26;
+  // True if RDS enhanced metrics are disdabled.
+  bool rds_enhanced_metrics_disabled = 27;
+  // Node identifier where this instance runs.
+  string runs_on_node_id = 28;
+  // Service identifier.
+  string service_id = 29;
+  // Actual Agent status.
+  string status = 30;
+  // Last known table count.
+  int32 table_count = 31;
+  // Tablestats group collectors are disabled if there are more than that number of tables.
+  // 0 means tablestats group collectors are always enabled (no limit).
+  // Negative value means tablestats group collectors are always disabled.
+  int32 table_count_tablestats_group_limit = 32;
+  // Use TLS for database connections.
+  bool tls = 33;
+  // Skip TLS certificate and hostname validation.
+  bool tls_skip_verify = 34;
+  // HTTP basic auth username for collecting metrics.
+  string username = 35;
+  // Last update timestamp.
+  google.protobuf.Timestamp updated_at = 36;
+  // Agent version.
+  string version = 37;
+  // True if Agent is running and connected to pmm-managed.
+  bool is_connected = 38;
+}
+
+message ListAgentRequest {
+  // Return only Agents that relate to a specific ServiceID.
+  string service_id = 1 [(validator.field) = {string_not_empty: true}];
```

Comment:
> I think we also should include service_name because it's easier to use for end-users.

It wasn't required so far, unless we want to offer an additional convenience.

Reply:
it's management API, management API should be more high-level API compared to inventory API.

Reply:
Will do in the GA version of this API.